src/lennybot/check/docker_image_available.py

```diff
@@ -66,20 +77,61 @@ def _parse_image(self):
             return DockerImage(None, match.group(2) + "/" + match.group(3), image_tag)
         return DockerImage(match.group(4), match.group(5) + "/" + match.group(6), image_tag)
 
+    def _authenticate_on_registry(self, registry: str, realm: str, service: str, scope: str) -> str:
+        if registry not in self._container_config.registries.keys():
+            raise Exception(f"No credentials found for registry: {registry}")
+
+        registry_data = self._container_config.registries[registry]
+
+        if registry_data.password is None or "":
```

Comment:
The handling of registries without credentials should not require a registry defined in the configuration file.